### PR TITLE
Endpoints are now checked multiple times and max 30 seconds

### PR DESCRIPTION
**Pull request by a-breid-ericsson**

### CODE REVIEW

File: src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java [hunk 1]
```diff
@@ -153,10 +154,19 @@ public void check_mail_subscriptions_were_triggered() throws Throwable {
     @Then("^Rest subscriptions were triggered$")
     public void check_rest_subscriptions_were_triggered() throws Throwable {
         LOGGER.debug("Verifying REST requests.");
-        assert (requestBodyContainsStatedValues(REST_ENDPOINT));
-        assert (requestBodyContainsStatedValues(REST_ENDPOINT_AUTH));
-        assert (requestBodyContainsStatedValues(REST_ENDPOINT_PARAMS));
-        assert (requestBodyContainsStatedValues(REST_ENDPOINT_AUTH_PARAMS));
+        List<String> endpointsToCheck = new ArrayList<String>() {
```
Comment (ned29) [inline src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java]: make it shorter 
List< String > endpointsToCheck = new ArrayList<>(Arrays.asList(REST_ENDPOINT,REST_ENDPOINT_AUTH,.....))

File: src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java [hunk 2]
```diff
@@ -214,6 +224,35 @@ private void validateSubscriptionsSuccessfullyAdded(String endPoint) throws Exce
...
+     * @param expectedCalls
+     *            Integer with the least number of calls.
+     * @return true if all endpoints had atleast the number of calls as
+     *         expected.
+     * @throws JSONException
+     * @throws InterruptedException
+     */
+    private boolean allEndpointsGotAtLeastOneCall(final List<String> endpoints, int expectedCalls)
+            throws JSONException, InterruptedException {
+        List<String> endpointsToCheck = new ArrayList<String>(endpoints);
```
Comment (ned29) [inline src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java]: remove String it's diamonds type   
List< String > endpointsToCheck = new ArrayList<>(endpoints);

File: src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java (hunk 1 above)
Comment (a-breid-ericsson) [reply src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java]: Done

File: src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java (hunk 2 above)
Comment (a-breid-ericsson) [reply src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java]: Done.

File: src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java [hunk 3]
```diff
@@ -214,6 +220,35 @@ private void validateSubscriptionsSuccessfullyAdded(String endPoint) throws Exce
...
+     *            List of endpoints to check.
+     * @param expectedCalls
+     *            Integer with the least number of calls.
+     * @return true if all endpoints had atleast the number of calls as
+     *         expected.
+     * @throws JSONException
+     * @throws InterruptedException
+     */
+    private boolean allEndpointsGotAtLeastXCalls(final List<String> endpoints, int expectedCalls)
+            throws JSONException, InterruptedException {
```
Comment (tobiasake) [inline src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java]: Where is the unit tests for this function?

Comment (tobiasake) [reply src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java]: I guess no unit tests has been created for the other functions. So I guess I can approve this change for the moment.
But we should have unit test implemented for the logics in these test base class and functions in a near future.

**Pull request merged**